demos/guide/quic-hq-interop-server.c

```diff
@@ -332,6 +332,77 @@ static BIO *create_socket(uint16_t port)
     return NULL;
 }
 
+/**
+ * @brief Handles I/O failures on an SSL stream based on the result code.
+ *
+ * This function processes the result of an SSL I/O operation and handles
+ * different types of errors that may occur during the operation. It takes
+ * appropriate actions such as retrying the operation, reporting errors, or
+ * returning specific status codes based on the error type.
+ *
+ * @param ssl A pointer to the SSL object representing the stream.
+ * @param res The result code from the SSL I/O operation.
+ * @return An integer indicating the outcome:
+ *         - 1: Temporary failure, the operation should be retried.
+ *         - 0: EOF, indicating the stream has been closed.
+ *         - -1: A fatal error occurred or the stream has been reset.
+ *
+ *
+ * @note If the failure is due to an SSL verification error, additional
+ * information will be logged to stderr.
+ */
+static int handle_io_failure(SSL *ssl, int res)
+{
+    switch (SSL_get_error(ssl, res)) {
+    case SSL_ERROR_WANT_READ:
+    case SSL_ERROR_WANT_WRITE:
+        return 1;
```

Comment:
Since we're using blocking IO, we don't really expect to see `SSL_ERROR_WANT_READ` or `SSL_ERROR_WANT_WRITE` here. If we get these doesn't this indicate some problem in our blocking implementation? We could just leave this as is (but perhaps insert a comment)...or alternatively remove these and let the "default" case handle this as a "failure".